### DISCUSSION TITLE
Customize TextBox colors

### DISCUSSION THREAD

**Discussion by ahpxex** [AvaloniaCommunity/Material.Avalonia#383] [Q&A] [answered]
This is a TextBox:
```xml
<TextBox Text="zzz"  VerticalAlignment="Top"/>
```
But its color is inherited from the `PrimaryColor` and I want to customize which, what should I do?
![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/34391004/d387ac66-66fb-4fb7-a800-bcf128e85ff8)

**Comment by SKProCH** [comment]
Hello, @SinoAHpx 

You can directly set colors, e.g.:
```xaml
<TextBox Background="Blue" />
```
Or, alternative way is to override a resource values which TextBox uses.

**Comment by ahpxex** [comment]
Original: 
```xml
<TextBox Text="AWDAWD" />
```
![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/34391004/2eb96190-56a3-436b-a76c-a519602c629c)

With `Background` attribute set:
```xml
 <TextBox Background="Indigo" Text="AWDAWD"/>
```
![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/34391004/615ca900-4aac-46d2-82c6-44a375f1f44b)

**Reply by SKProCH** [reply]
So, is it working fine?

**Reply by ahpxex** [reply]
Unfortunately, as you can see, setting the background working terribly

**Reply by SKProCH** [reply]
So, what do you expect from this?

**Reply by ahpxex** [reply]
Oh, I apologize that I didn't explain clearly. 
For example, the effect below is implemented only by changing `PrimaryColor` in `App.axaml`, I wonder if there's another way to do the same
![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/34391004/26ceef34-8ef5-4005-8782-974ddcf6f5d6)

**Reply by SKProCH** [reply, marked as answer]
So, you want override a PrimaryColor for a certain control only? You can actually do it by overriding a resource near the control (Avalonia looks for resources starting from the control and climbing up in visual tree, App resources used when nothing found previously). So, you can do something like this:
```xaml
<TextBox ...>
  <TextBox.Resources>
    <SolidColorBrush x:Key="MaterialTextBoxBorderBrush" Color="Red" />
    <SolidColorBrush x:Key="MaterialPrimaryMidBrush" Color="Red" />
  </TextBox.Resources>
</TextBox>
```
Default style for TextBox underline utilizes the following keys: `MaterialTextBoxBorderBrush` and `MaterialPrimaryMidBrush`.

**Reply by ahpxex** [reply]
I sincerely appreciate your answer. Have a nice day!
